src/content.js

```diff
@@ -45,22 +45,75 @@ Function used to avoid multiple injection (cleaner than using an if?)
     var showTranslated = true;
     var orientation = 'horizontal-tb';
 
+    // Set to track successfully processed images to avoid reprocessing
+    const processedImages = new WeakSet();
+    // Set to track images currently being processed to avoid concurrent processing
+    const processingImages = new WeakSet();
+
+    // Helper function to check if a node is an image
+    function isImageNode(node) {
+        return node.nodeType === Node.ELEMENT_NODE && 
+               (node.tagName === 'IMG' || node.tagName === 'CANVAS');
+    }
+
+    // Helper function to process all images in a subtree
+    function processImagesInSubtree(root) {
+        if (isImageNode(root) && !processedImages.has(root)) {
+            debug('new image found in subtree', root);
+            processImage(root);
+            return;
```

Comment:
```suggestion
        if (isImageNode(root)) {
            if (!processedImages.has(root)) {
                debug('new image found in subtree', root);
                processImage(root);
            }
            return;
```

Avoid running the subtree query if a node is already an image